src/tox/config/source/toml.py

```diff
@@ -0,0 +1,142 @@
+"""Support for TOML config sources.
+
+This is experimental API! Expect things to be broken.
+"""
+
+from __future__ import annotations
+
+from collections import defaultdict
+from itertools import chain
+from typing import TYPE_CHECKING, Iterable, Iterator
+
+import tomllib
+
+from tox.config.loader.ini.factor import find_envs
+from tox.config.loader.section import Section
+from tox.config.loader.toml import TomlLoader
+
+from .api import Source
+from .ini_section import CORE, PKG_ENV_PREFIX, TEST_ENV_PREFIX, IniSection
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from tox.config.loader.api import OverrideMap
+    from tox.config.sets import ConfigSet
+
+
+class TomlSource(Source):
+    """Configuration sourced from a toml file (such as tox.toml).
+
+    This is experimental API! Expect things to be broken.
+    """
+
+    CORE_SECTION = CORE
+
+    def __init__(self, path: Path, content: str | None = None) -> None:
+        super().__init__(path)
+        if content is None:
+            if not path.exists():
+                raise ValueError
+            content = path.read_text()
+        self._raw = tomllib.loads(content)
+        self._section_mapping: defaultdict[str, list[str]] = defaultdict(list)
+
+    def __repr__(self) -> str:
+        return f"{type(self).__name__}(path={self.path})"
+
+    def transform_section(self, section: Section) -> Section:  # noqa: PLR6301
+        return IniSection(section.prefix, section.name)
```

Comment:
We should not use the ini system at all.

Reply:
I duplicated `IniSection` to `TomlSection` for the time being. I plan to do more clean up later on.